### PR TITLE
TA-1744 fix flaky stage billing bulkload test

### PR DESCRIPTION
**Pull request by solomonahmed**
## What does this pull request do?

Please describe what you did.
2 stage billing validation feature tests have been moved to a new stage_billing_manual_validation.feature.  Creating the necessary submissions manually seems to work and only adds around 4 secs to the test (the feature test is very simple).

To support this work a number of other changes were made:
The cwa_outcome_steps.rb file was changed to ensure the ScreenBuilder was called with either a claim_type _or_ matter_type for Legal Help or Crime Lower respectively.  This is so it can locate the correct set of fields/defaults in the relevant yaml file.  Before it was defaulting to 'completed_matter' always for Legal Help which was causing a problem when the tests needed to grab stage claim or disbursement claim fields from the yaml file.  For Crime Lower the matter_type wasn't being passed at all.  It's been renamed from stage_reached_code to matter_type in the class to more accurately reflect the business domain name.

Yaml files where updated to fix a number of problems with the ScreenBuilder expecting fields to populate which weren't on the screen.

Removed 2 feature tests from previous bulkload test feature for stage billing.  There are still some failing tests in this file which need to be addressed (maybe as part of this work?).  Need to consult first as it seems the bulkload builder is not building the bukload correctly.  Unsure whether to fix this or port the tests to a manual test?

## Why make these changes?

Please describe why the changes were needed.
The stage billing validation feature for Mental Health was failing.  After much investigation it seemed to be the result of a general flakiness in the UI where a submission wasn't being saved.  This initial submission needed to be saved for the subsequent test to complete successfully.  Rather than delve deeper into solving this after a team discussion we decided to port the bulkload tests to manual ones.

## Checklist

- [x] Provided link to a JIRA ticket: https://dsdmoj.atlassian.net/browse/TA-1744

### CODE REVIEW

File: features/step_definitions/cwa_outcome_steps.rb [hunk 1]
```diff
@@ -24,16 +24,26 @@
 end
 
 When("user adds outcomes for {string} {string} with fields like this:") do |area_of_law, category_of_law, table|
+  lh_claim_type = { "DC" => "disbursement_claim", "SC" => "stage_claim" }
+  lh_claim_type.default = "completed_matter"
```
Comment (your) [inline features/step_definitions/cwa_outcome_steps.rb:28]: I would move this to `features/support/helpers/screen_field_builder.rb`

```ruby
...
    class LegalHelp < Base
      CLAIM_TYPES = {
        'DC' => 'disbursement_claim',
        'SC' => 'stage_claim',
        'CM' => 'completed_matter'
      }
      def fields
        ...

           private

      def claim_type
        CLAIM_TYPES[extra_args.fetch(:claim_type, 'CM').to_s]
      end

       ...
...
```

File: features/step_definitions/cwa_outcome_steps.rb [hunk 2]
```diff
@@ -24,16 +24,26 @@
...
   outcome_data.each do |test_data_row|
     submission_list_page = SubmissionListPage.new
     submission_list_page.add_outcome_button.click
     test_data_row["schedule_ref"] = sr
+    if area_of_law == 'Legal Help'
+      claim_type = lh_claim_type[test_data_row["claim_type"]]
+    end
+    if area_of_law == 'Crime Lower'
+      matter_type = test_data_row["matter_type"]
+    end
```
Comment (your) [inline features/step_definitions/cwa_outcome_steps.rb:41]: I would always pass `claim_type` and `matter_type`; if they are `nil` and the builder doesn't actually need them, no harm done

File: features/step_definitions/cwa_outcome_steps.rb [hunk 3]
```diff
@@ -24,16 +24,26 @@
...
     builder = Helpers::ScreenFieldBuilder.from(
       area_of_law: area_of_law.downcase.gsub(' ', '_'),
-      category_of_law: category_of_law.downcase.gsub(' ', '_')
+      category_of_law: category_of_law.downcase.gsub(' ', '_'),
+      claim_type: claim_type,
+      matter_type: matter_type
     )
```
Comment (your) [inline features/step_definitions/cwa_outcome_steps.rb:47]: I'd rewrite this step quickly with:

```ruby
  outcome_data = table.hashes
  outcome_data.each do |outcome|
    submission_list_page = SubmissionListPage.new
    submission_list_page.add_outcome_button.click

    builder = Helpers::ScreenFieldBuilder.from(
      area_of_law: area_of_law.downcase.gsub(' ', '_'),
      category_of_law: category_of_law.downcase.gsub(' ', '_'),
      claim_type: outcome['claim_type'],
      matter_type: outcome['matter_type']
    )
    outcome['schedule_ref'] = CWAProvider.submission.schedule_ref
    builder.overrides = outcome

    page = AddOutcomePage.new(builder)
    page.add_outcome
  end

  @submissions_saved = outcome_data.size
```

To make it a tad more readable for now

**Review by your — CHANGES_REQUESTED** [review]
Great work, as usual, and thank you for spotting the bug in the form builder!

Requesting changes mainly because a step is being skipped due to a typo.

Other comments are code style-related to increase readability; let me know what you think.

**Comment by solomonahmed** [comment]
Thanks for review @your .  Changes made as suggested.  Please re-review when you get a chance.

**Review by your — APPROVED** [review]
💯

**Pull request merged**